cookbooks/bcpc-hadoop/templates/default/graphite.query_graphite.config.erb

```diff
@@ -1,17 +1,23 @@
 {
- "queries":[
-<% node[:bcpc][:hadoop][:graphite][:queries].each_with_index do |(host, value),hidx| -%>
-  <% value.each_with_index do |q,aidx| -%>
-    <% if hidx == node[:bcpc][:hadoop][:graphite][:queries].length-1 and aidx == value.length()-1 -%>
-      {"graphite_url": "https://<%=node[:bcpc][:management][:vip]%>:<%=node[:bcpc][:graphite][:web_port]%>/render/?from=-2min&target=<%=q['type']%>.<%= host %>.*.<%=q['query']%>&format=json",
-       "zabbix_host" : "<%= host %>", 
-       "zabbix_key"  : "<%=q['key']%>"}
-    <% else -%>
-      {"graphite_url": "https://<%=node[:bcpc][:management][:vip]%>:<%=node[:bcpc][:graphite][:web_port]%>/render/?from=-2min&target=<%=q['type']%>.<%= host %>.*.<%=q['query']%>&format=json",
-       "zabbix_host" : "<%= host %>", 
-       "zabbix_key"  : "<%=q['key']%>"},
-    <% end %>
-  <% end %>
-<% end %>
- ]
+  "queries":[
+  <% node["bcpc"]["hadoop"]["graphite"]["queries"].each_with_index do |(host, queries),hidx| %>
+    <% queries.each_with_index do |(key, attrs),qidx| %>
+      <% if attrs['type'] == 'jmx' %>
+        {"graphite_url": "https://<%=node[:bcpc][:management][:vip]%>:<%=node[:bcpc][:graphite][:web_port]%>/render/?from=-2min&target=<%=attrs['type']%>.<%=host%>.*.<%=attrs['query']%>&format=json",
+      <% elsif attrs['type'] == 'chef' %>
+        {"graphite_url": "https://<%=node[:bcpc][:management][:vip]%>:<%=node[:bcpc][:graphite][:web_port]%>/render/?from=-2min&target=<%=attrs['type']%>.<%=host%>.<%=attrs['query']%>&format=json",
+      <% elsif attrs['type'] == 'function' %>
+        {"graphite_url": "https://<%=node[:bcpc][:management][:vip]%>:<%=node[:bcpc][:graphite][:web_port]%>/render/?from=-2min&target=<%=attrs['query']%>&format=json",
+      <% else #servers %>
+        {"graphite_url": "https://<%=node[:bcpc][:management][:vip]%>:<%=node[:bcpc][:graphite][:web_port]%>/render/?from=-2min&target=<%=attrs['type']%>.<%=host%>.<%=attrs['query']%>&format=json",
+      <% end %>
+        "zabbix_host" : "<%=host%>",
+      <% if hidx == node[:bcpc][:hadoop][:graphite][:queries].length-1 and qidx == queries.length()-1 %>
+        "zabbix_key" : "<%=key%>"}
+      <% else %>
+        "zabbix_key" : "<%=key%>"},
```

Comment:
Erm, might be easier to just make this all a Ruby hash and call the JSON gems' [generate](http://ruby-doc.org/stdlib-2.0.0/libdoc/json/rdoc/JSON.html#module-JSON-label-Generating+JSON).